### ISSUE TITLE
Doesn't work on iOS

### ISSUE THREAD

**Issue by BenoitDuffez** [nimiq/qr-scanner#121]
I have tested a lot of Android devices, including old original Moto G or more recent Pixel 3XL, tablets: they all work.

I have only 2 iOS devices: iPhone 6S and iPad 5. They don't work. Some customers/partners tried with other iOS devices: they will not scan the QR.

Admittedly, it's small, but Android has no issue scanning. What's even weirder is that when I try to scan with the iPad, the QR is so clear on screen that my Pixel 3XL is able to scan the QR displayed on the iPad screen! Meaning that the picture the iPad has is just the right quality and it should work.

The Camera app on all iOS devices is able to instantly recognize the QR code. It just doesn't work on the web using your lib. Tested on Safari & Chrome.

**Comment by sisou** [comment]
Hi, thanks for testing and reporting!

What iOS versions do your iDevices run?

**Comment by BenoitDuffez** [comment]
Thanks for your feedback. They all use the latest version available today.

**Comment by sisou** [comment]
Great, now I should look up what the latest supported version for an iPhone 6S is? Just please go into your setting and list the version.

**Comment by BenoitDuffez** [comment]
Oh, no, sorry about that. Latest as in latest, they're all on iOS 14.7.

**Comment by BenoitDuffez** [comment]
Is there any trace or other info I can provide that would help?

**Comment by sisou** [comment]
Sorry for the delay.
Would you be able to provide an example QR code that works on your Android devices, but fails to scan under iOS?

**Comment by danimoh** [comment]
Note that Android and iOS use different implementations, therefore it working on Android unfortunately doesn't tell much about iOS yet.
The same implementation is shared between iOS and desktop browsers though (with the exception of Webkit based browsers on Mac like Chrome or Opera).
Please check whether you're also experiencing issues on desktop.
If you could provide us with an example QR code as @sisou said and with a stack trace it would be appreciated :+1:

**Comment by BenoitDuffez** [comment]
Sorry for the delay. Here's the QR:

    $ qrencode -v 1 -8 -o /tmp/qr.png '735BRc030010'

![qr](https://user-images.githubusercontent.com/802209/134042476-2c89968e-cd7e-424b-84e7-28dd7004a5d9.png)

The printed size is about 5x5 millimeters. I can send a picture of it but it's not really relevant I guess.

Re: desktop  
I tested on Chrome on Linux, but the camera itself is crappy (like blurry). I think I still managed to scan it because even if the demo shows "No QR code detected", it shows "Last detected at" with a timestamp. I'm assuming it worked.  
I tested on Chrome on macOS, it never worked, despite the image being clearer than on my Linux laptop.

I don't have more computers to test on (and I don't have any running Windows if that would matter).

**Comment by BenoitDuffez** [comment]
I could also test Safari on mac, and Firefox on both Linux/Mac if that would help/make a difference.

**Comment by knufri** [comment]
@BenoitDuffez did you include the setting up the worker like in the documentation? I first used the qr scanner without that and it also didn't work on iOS. But after including the worker (qr-scanner-worker.min.js) it works on my native iPhone 12 with the latest iOS15 also on my Safari 15 on MacOS Big Sur (11.5.2).

**Comment by BenoitDuffez** [comment]
Yes I did, but I may have done it wrong. I'm using webpack and I have this in a .js file included in webpack:

```js
import QrScanner from 'qr-scanner';
import QrScannerWorkerPath from '!!file-loader!../node_modules/qr-scanner/qr-scanner-worker.min.js';

QrScanner.WORKER_PATH = QrScannerWorkerPath;
```

Is there a way to 100% confirm the worker is loaded and works?

**Comment by ChrisCosentino** [comment]
I am having the same issue I get a `Scanner error: timeout` whenever I scan a qr code only on ios 14.7.1, testing on my laptop works. 

I am using webpack and these are my imports: 
```
import QRScanner from 'qr-scanner'
import QRScannerWorkerPath from 'qr-scanner/qr-scanner-worker.min';

QRScanner.WORKER_PATH = QRScannerWorkerPath;
```

The `WORKER_PATH` is found because when I remove it, I get an error that it is not found

**Comment by ksinas** [comment]
I also have the same `Scanner error: timeout` issue. Worker is loaded correctly.

**Comment by iveqy** [comment]
I have the same error on my iPhone, on my MacBook Air m1, it works fine with firefox though:
Scanner error: timeout

**Comment by ksinas** [comment]
> I have the same error on my iPhone, on my MacBook Air m1, it works fine with firefox though:
> Scanner error: timeout

You need to make sure the worker is loaded correctly. Safari doesn't have support for native BarcodeDetector api

**Comment by danimoh** [comment]
Do you guys still have these issues with the newest release?

Note that with the newest release, specifying the WORKER_PATH is not required anymore, see readme.
